src/Test/ItemsTest.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentApiBundle\Test;
+
+use AppendIterator;
+use Libero\ContentApiBundle\Exception\ItemNotFound;
+use Libero\ContentApiBundle\Exception\UnexpectedVersionNumber;
+use Libero\ContentApiBundle\Exception\VersionNotFound;
+use Libero\ContentApiBundle\Model\ItemId;
+use Libero\ContentApiBundle\Model\ItemListPage;
+use Libero\ContentApiBundle\Model\Items;
+use Libero\ContentApiBundle\Model\ItemVersion;
+use Libero\ContentApiBundle\Model\ItemVersionNumber;
+use PHPUnit\Framework\TestCase;
+use function fopen;
+use function fwrite;
+use function iterator_to_array;
+use function rewind;
+
+abstract class ItemsTest extends TestCase
```

Comment:
Since this is abstract it could avoid the `Test` suffix like `TestCase` classes do (`ItemsTestCase`) but being fancy on terminology we could specialize that into `ItemsContract` (slightly similar to what described in https://martinfowler.com/bliki/ContractTest.html but without integration)

Reply:
Will use `TestCase`. `Contract` sounds good, but technically other implementations also adhere to the contract (eg `NullItems`) but wouldn't pass this. Suggests that a prefix is needed on `ItemsTestCase`, but will leave for now.